test/unit_tests/cellular/cellular.cpp

```diff
@@ -78,6 +99,64 @@ TEST_CASE("cellular_signal()") {
         }
     }
 
+    SECTION("CellularSignalHal RSSI and QUAL") {
+        NetStatus status;
+        memset(&status, 0, sizeof(status));
+        status.rssi = -50; // arbitrary non-zero values
+        status.qual = 37;  //  "
+
+        SECTION("CellularSignalHal::rssi and CellularSignalHal::qual are set to expected values") {
+            CellularSignalHal signal;
+            memset(&signal, 0, sizeof(signal));
+            REQUIRE(cellular_signal_impl(&signal, nullptr, true, status) == SYSTEM_ERROR_NONE);
+            REQUIRE(signal.rssi == -50);
+            REQUIRE(signal.qual == 37);
+        }
+    }
+
+    SECTION("UNKNOWN") {
+        NetStatus status;
+        memset(&status, 0, sizeof(status));
+        status.act = ACT_UNKNOWN;
+
+        SECTION("values reported by ACT_UNKNOWN case") {
+            status.rxlev = 10; // seed all values as 10 just to ensure default case is hit
+            status.rxqual = 10;
+            status.rscp = 10;
+            status.ecno = 10;
+            status.rsrp = 10;
+            status.rsrq = 10;
+
+            cellular_signal_t sig = {0};
+            sig.size = sizeof(sig);
+            REQUIRE(cellular_signal_impl(nullptr, &sig, true, status) == SYSTEM_ERROR_UNKNOWN);
```

Comment:
This test seems naked (i.e. it should have it's own `WHEN` `THEN`).

Reply:
This is really more of a THEN that is always done before CellularSignal outputs are tested.  Can't wrap it in another nested SECTION or else it won't work correctly.